workspaces/orchestrator/docs/extensibleForm.md

```diff
@@ -0,0 +1,136 @@
+  apis: [formApiFactory],
+});
+```
+
+### Schema example for above plugin
+
+```typescript
+{
+  "type": "object",
+  "properties": {
+    "personalDetails": {
+      "type": "object",
+      "title": "Personal Details",
+      "properties": {
+        "name": {
+          "type": "string",
+          "title": "Name"
+        },
+        "country": {
+          "type": "string",
+          "title": "Country",
+          "ui:widget": "CountryWidget"
+        }
+      }
+    },
+    "contactDetails": {
+      "type": "object",
+      "title": "Contact Details",
+      "properties": {
+        "email": {
+          "type": "string",
+          "title": "Email"
+        },
+        "phone": {
+          "type": "string",
+          "title": "Phone Number"
+        }
+      }
+    }
+  }
+}
+```
+
+### Dynamic plugin configuration example
+
+Add the following to backstage config to integrate the plugin:
+
+```yaml
+dynamicPlugins:
+  frontend:
+    custom-form-plugin:
+      apiFactories:
+        - importName: formApiFactory
+```
+
+### Referencing the custom behavior in the schema
+
+The workflow execution schema adheres to the [json-schema](https://json-schema.org/) format, which allows for extending the schema with custom properties beyond the official specification. This flexibility enables the inclusion of additional [uiSchema](https://rjsf-team.github.io/react-jsonschema-form/docs/api-reference/uiSchema/) fields directly within the schema, as demonstrated in the example above.
+
+### How It All Comes Together
+
+The `orchestrator-form-react` plugin implements the form component for workflow execution. It integrates with the custom API provided by the developer's plugin to generate and customize the form. The `orchestrator` plugin then incorporates this form into the workflow execution page.
+
+The `orchestrator-form-react` plugin handles the following key tasks:
+
+- **Generating the UI Schema:** It extracts custom UI schema fields from the main schema, automatically generates the [uiSchema](https://rjsf-team.github.io/react-jsonschema-form/docs/api-reference/uiSchema/), and passes it to the `react-jsonschema-form` component, enabling advanced UI customizations.
+
+- **Organizing Forms into Wizard-Style Steps:** If the schema is an object containing nested objects (i.e., the root is an object, and its properties are also objects), the plugin organizes the form into multiple steps. Each nested object becomes a separate step in a wizard-style interface. For example, the schema provided above results in two steps: _Personal Details_ and _Contact Details_.
+
+The [`orchestrator-form-react`](https://github.com/janus-idp/backstage-plugins/tree/main/plugins/orchestrator-form-react) plugin is designed to operate independently of the main orchestrator plugin. This modularity allows developers to test and validate form behavior in a standalone Backstage development environment before integrating it with the full orchestrator setup.
```

Comment:
https://github.com/janus-idp/backstage-plugins/tree/main/plugins/orchestrator-form-react is depracated.